project/src/main/kotlin/org/ionproject/core/calendar/icalendar/types/Duration.kt

```diff
@@ -0,0 +1,127 @@
+package org.ionproject.core.calendar.icalendar.types
+
+abstract class Duration private constructor(
+    val adding: Boolean
```

Comment:
The `adding` property doesn't have anything to do it is just because the duration's value can be negative. It is used commonly for alarms. From the RFC:
> Negative durations are typically used to schedule an alarm to
      trigger before an associated time (see Section 3.8.6.3).